### PR TITLE
Use signatures in most of testapi (test covered changes)

### PR DESCRIPTION
**Pull request by okurz**
* Simplify testapi::check_var
* Use proper perl "signatures" in most of testapi
* t: Add test for testapi::x11_start_program
* t: Add test for set_var, get_var_array, check_var_array
* t: Add test for testapi::force_soft_failure
* t: Add test for testapi::init

I created this PR out of
* #1696

including only the changes that are covered by automatic tests.

### CODE REVIEW

**Review by Martchus — APPROVED** [review]

**Comment by Martchus** [comment]
The style check is still failing, though.

**Comment by okurz** [comment]
> The style check is still failing, though.

fixed

File: testapi.pm [hunk 1]
```diff
@@ -197,23 +192,21 @@ influenced.
...
     bmwqemu::log_call(reason => $reason);
 
     $autotest::current_test->record_soft_failure_result($reason);
 }
 
-sub _is_valid_result {
-    my ($result) = @_;
+sub _is_valid_result ($result) {
     return $result =~ /^(ok|fail|softfail)$/;
 }
```
Comment (kalikiana) [inline testapi.pm:203]:
```suggestion
sub _is_valid_result ($result) { $result =~ /^(ok|fail|softfail)$/ }
```

File: testapi.pm [hunk 2]
```diff
@@ -477,11 +463,8 @@ been matched at the time of the call.
...
 =cut
 
-sub match_has_tag {
-    my ($tag) = @_;
-    if ($last_matched_needle) {
-        return $last_matched_needle->{needle}->has_tag($tag);
-    }
+sub match_has_tag ($tag) {
+    return $last_matched_needle->{needle}->has_tag($tag) if $last_matched_needle;
     return;
```
Comment (kalikiana) [inline testapi.pm:468]:
```suggestion
    return undef;
```

Might as well fix the return value here. We should always return explicitly

File: testapi.pm (hunk 1 above)
Comment (okurz) [reply testapi.pm:203]: I didn't want to change something unrelated to signatures

Comment (okurz) [reply testapi.pm]: ok, changed

File: testapi.pm (hunk 2 above)
Comment (okurz) [reply testapi.pm]: done

**Review by kalikiana — APPROVED** [review]

File: testapi.pm [hunk 3]
```diff
@@ -1559,9 +1530,9 @@ Default hold time is 0.15s
 
 =cut
 
-sub mouse_click {
-    my $button = shift || 'left';
-    my $time = shift || 0.15;
+sub mouse_click ($button = undef, $time = undef) {
+    $button //= 'left';
+    $time //= 0.15;
```
Comment (perlpunk) [inline testapi.pm:1529]: Strictly, if `$time` was 0, then it would have been set to `0.15` in the original code. Now only if it's undef.

Comment (perlpunk) [reply testapi.pm:1529]: ok, I just wanted to mention it.

**Pull request merged**